package_bag/routines.py

```diff
@@ -0,0 +1,211 @@
+        unprocessed = self.discover_bags(settings.SRC_DIR)
+        for u in unprocessed:
+            try:
+                bag_id = self.unpack_rename(u, settings.TMP_DIR)
+                bag = Bag.objects.create(
+                    original_bag_name=u,
+                    bag_identifier=bag_id,
+                    bag_path=join(
+                        settings.TMP_DIR,
+                        bag_id))
+                self.validate_structure(bag.bag_path)
+                self.validate_metadata(bag.bag_path)
+                self.get_data(bag)
+                processed.append(bag)
+            except Exception as e:
+                print(e)
+        # what does this process bags function return? - you want to return something out of the view that indicates which objects were processed
+        return processed
+
+    def discover_bags(self, src):
+        """Looks in a given directory for compressed bags, adds to list to process"""
+        bags_list = []
+        for d in listdir(src):
+            ext = splitext(d)[-1]
+            if ext in ['.tgz', '.tar.gz', '.gz']:
+                bags_list.append(join(src, d))
+        return bags_list
+
+    def unpack_rename(self, bag_path, tmp):
+        """Unpacks tarfile to a new directory with the name of the bag identifier (a UUID)"""
+        bag_identifier = str(uuid4())
+        tf = tarfile.open(bag_path, 'r')
+        tf.extractall(tmp)
+        original_bag_name = tf.getnames()[0]
+        tf.close()
+        rename(join(tmp, original_bag_name),
+               join(tmp, bag_identifier))
+        remove(bag_path)
+        return bag_identifier
+
+    def validate_structure(self, bag_path):
+        """Validates a bag against the BagIt specification"""
+        new_bag = bagit.Bag(bag_path)
+        return new_bag.validate()
+
+    def validate_metadata(self, bag_path):
+        """Validates the bag-info.txt file against the bagit profile"""
+        # TO DO: first vlaidation that "BagIt-Profile-Identifier" exists
+        new_bag = bagit.Bag(bag_path)
+        bag_info = new_bag.info
+        if "BagIt-Profile-Identifier" not in bag_info:
+            print("no bagit profile identifier")
+            # TO DO: return exception
+        else:
+            profile = bagit_profile.Profile(
+                new_bag.info.get("BagIt-Profile-Identifier"))
+            # TO DO: exception if cannot retrieve profile
+            return profile.validate_bag_info(new_bag)
+            # TO DO: exception if validation does not work
+
+    def get_data(self, bag):
+        """Saves bag data from the bag-info.txt file"""
+        new_bag = bagit.Bag(bag.bag_path)
+        bag.origin = new_bag.info.get('Origin')
+        bag.rights_id = new_bag.info.get('Rights-ID')
+        bag.end_date = new_bag.info.get('End-Date')
+        bag.save()
+        # TO DO: what is this returning?
+
+
+# how does something get sent from one bag to another? how does batching work?
+class GetRights(object):
```

Comment:
```suggestion
class RightsAssigner(object):
```
See comment above about class names. Also this is doing more than just getting rights, it's also saving data about them to an object.